static/js/publisher/builds/index.js

```diff
@@ -84,6 +101,30 @@ class Builds extends React.Component {
     }
   }
 
+  triggerBuildHandler() {
+    const { csrf_token, snapName } = this.props;
+    const url = `/${snapName}/builds/trigger-build`;
+
+    this.setState({ triggerBuildLoading: true });
+
+    fetch(url, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        "X-CSRFToken": csrf_token
+      }
+    })
+      .then(res => res.json())
+      .then(result => {
+        if (result.success) {
+          this.setState({ triggerBuildSucccess: true });
+        }
+      })
+      .catch(() => {
+        this.setState({ triggerBuildSucccess: "ERROR" });
```

Comment:
we should define some consts for different states
```
const ERROR = "ERROR";
const SUCCESS = "SUCCESS";
const IDLE = "IDLE"
```

And use those, rather than a mix of boolean, string and null types.